core/domain/user_services_test.py

```diff
@@ -1302,6 +1302,26 @@ def test_add_user_role_to_mobile_learner_raises_exception(self) -> None:
             user_services.add_user_role(
                 profile_user_id, feconf.ROLE_ID_FULL_USER)
 
+    def test_add_duplicate_user_role_raises_exception(self) -> None:
+        auth_id = 'test_id'
+        user_email = 'test@gmail.com'
+
+        user_id = user_services.create_new_user(auth_id, user_email).user_id
+        self.assertEqual(
+            user_services.get_user_roles_from_id(user_id),
+            [feconf.ROLE_ID_FULL_USER])
+
+        user_services.add_user_role(
+            user_id, feconf.ROLE_ID_BLOG_POST_EDITOR)
+        self.assertEqual(
+            user_services.get_user_roles_from_id(user_id), [
+                feconf.ROLE_ID_FULL_USER, feconf.ROLE_ID_BLOG_POST_EDITOR])
+
+        error_msg = 'The user already has this role.'
```

Comment:
Minor nit: error_msg --> expected_error_msg

(it helps to be clear in tests what is expected vs observed)

Reply:
Done